backend/src/test/java/asegroup1/api/models/landregistry/LandRegistryQueryTestUtils.java

```diff
@@ -0,0 +1,240 @@
+		constraint.setMinDate(LocalDate.now().minusYears(5));
+		constraint.setMinPricePaid(150000);
+		constraint.setPostcodes(getRandomPostCodes());
+		return constraint;
+	}
+
+	static LandRegistryQuerySelect genLandRegistryQuerySelect() {
+		LandRegistryQuerySelect select = new LandRegistryQuerySelect();
+		for (Entry<Selectable, Aggrigation> entry : genRandomSelectableAggrigations().entrySet()) {
+			select.addSelectValue(entry.getKey(), entry.getValue());
+		}
+		return select;
+	}
+
+	static LinkedHashMap<Selectable, Aggrigation> genRandomSelectableAggrigations() {
+		ArrayList<Selectable> selectableSet = new ArrayList<>(EnumSet.allOf(Selectable.class));
+		ArrayList<Aggrigation> aggrigationSet = new ArrayList<>(EnumSet.allOf(Aggrigation.class));
+		Random rand = new Random(LandRegistryQueryTestUtils.randomSeed);
+
+		LinkedHashMap<Selectable, Aggrigation> map = new LinkedHashMap<>();
+
+		int iterations = rand.nextInt(selectableSet.size() - 4) + 3;
+		for (int i = 0; i < iterations; i++) {
+			Selectable selectable = selectableSet.remove(rand.nextInt(selectableSet.size()));
+			Aggrigation aggrigation = aggrigationSet.get(rand.nextInt(aggrigationSet.size()));
+			map.put(selectable, aggrigation);
+		}
+		return map;
+	}
+
+	static String[] getRandomPostCodes() {
+		return new String[] { "OX14 1WH", "L18 9SN", "TN27 8JG", "PL8 2EE" };
+	}
+
+	static List<Selectable> genRandomSelectables(Set<Selectable> options) {
+		Random r = new Random(LandRegistryQueryTestUtils.randomSeed);
+		ArrayList<Selectable> unSelected = new ArrayList<>(options);
+		ArrayList<Selectable> selected = new ArrayList<>();
+		int reps = r.nextInt(options.size() - 1) + 1;
+		for (int i = 0; i < reps; i++) {
+			selected.add(unSelected.remove(r.nextInt(unSelected.size())));
+		}
+
+		return selected;
+	}
+
+	static List<Selectable> genRandomSelectables() {
+		return genRandomSelectables(EnumSet.allOf(Selectable.class));
+	}
+
+	static String[] buildRandomSelectablesArray(List<Selectable> select) {
+		return select.stream().map(v -> v.toString()).collect(Collectors.toList()).toArray(new String[select.size()]);
+	}
+
+	static LandRegistryQueryGroup genLandRegistryQueryGroup() {
+		return new LandRegistryQueryGroup(buildRandomSelectablesArray(genRandomSelectables()));
+	}
+
+	/* REGEX */
+
+	static String buildQuerySelectRegex(boolean ignoreAggrigation) {
+		String start = "SELECT";
+
+		List<String> options = new ArrayList<>();
+		for (Selectable selectable : EnumSet.allOf(Selectable.class)) {
+			String str = selectable.toString();
+			options.add(str);
+			options.add(str.substring(0, 1).toUpperCase() + str.substring(1));
+		}
+
+		String optionList = regexOptionalList(options);
+		String option = "\\?" + optionList;
+
+		ArrayList<Aggrigation> aggrigations = new ArrayList<Aggrigation>(EnumSet.allOf(Aggrigation.class));
+		aggrigations.remove(Aggrigation.NONE);
+		
+		String aggrigationList = regexOptionalList(runOnList(aggrigations, v -> v.toString()));
```

Comment:
Can be replaced with `regexOptionalList(runOnList(aggrigations, Enum::toString))`